articles/actions.md

```diff
@@ -0,0 +1,338 @@
+In ROS 1 if a client submits a goal and immediatly tries to cancel it then the cancelation may or may not happen.
+If the cancelation is processed before the goal submission then `actionlib` will ignore the cancellation request without notifying the user's code.
+
+### Namespacing of Generated Messages and Services
+
+Multiple message and service definitions are generated from a single action definition.
+In ROS 1, the generated messages were prefixed with the name of the action to avoid conflicts with other messages and services.
+In ROS 2, the generated service and message definitions will exist in a different namespace to be impossible to conflict with non-action message and service definitions.
+For example, in Python the code from the generated definitions should be in the module `action` instead of `srv` and `msg`.
+In C++, the generated code should be in the namespace and folder `action` instead of `srv` and `msg`.
+
+### Visibility of Action Services and Topics
+
+In ROS 1 `rostopic list` would show all action topics in its output.
+In ROS 2 `ros2 topic list` and `ros2 service list` will not show topics and services used by actions by default.
+The can still be shown by passing an option to the commands to show hidden services and topics.
+
+## Action Interface Definition
+
+Actions are specified using a form of the ROS Message IDL.
+The specification contains three sections, each of which is a message specification:
+
+1. Goal
+
+   This describes what the action should achieve and how it should do it.
+   It is sent to the action server when it is requested to execute an action.
+
+1. Result
+
+   This describes the outcome of an action.
+   It is sent from the server to the client when the action execution ends, whether successfully or not.
+
+1. Feedback
+
+   This describes the progress towards completing an action.
+   It is sent to the client of the action from the action server between commencing action execution and prior to the action completing.
+   This data is used by the client to understand the progress of executing the action.
+
+Any of these sections may be empty.
+Between each of the three sections is a line containing three hyphens, `---`.
+Action specifications are stored in a file ending in `.action`.
+There is one action specification per `.action` file.
+
+### Example
+
+```
+# Define a goal of washing all dishes
+uint32 dishwasher_id  # Specify which dishwasher we want to use
+---
+# Define the result that will be published after the action execution ends.
+uint32 total_dishes_cleaned
+---
+# Define a feedback message that will be published during action execution.
+float32 percent_complete
+uint32 number_dishes_cleaned
+```
+
+## Introspection tools
+
+Actions, like topics and services, are introspectable from the command line.
+In ROS 1, actions are visible in the output of the `rostopic` tool.
+
+In ROS 2, actions will not be visible by default as a set of topics nor a set of services.
```

Comment:
Remove duplicate documentation here from *Visibility of Action Services and Topics*